### PR TITLE
Allow additional kwargs to pass from reproject_match() -> reproject()

### PR DESCRIPTION
**Pull request by four43**
I wanted to set a specific `nodata` value when running a `reproject_match` and I figured this was the most flexible. What do you all think?

 - [x] Tests added
 - [ ] Fully documented, including `docs/history.rst` for all changes and `docs/rioxarray.rst` for new API

### CODE REVIEW

File: rioxarray/raster_array.py [hunk 1]
```diff
@@ -456,7 +456,9 @@ def reproject(
         xda.rio.write_coordinate_system(inplace=True)
         return xda
 
-    def reproject_match(self, match_data_array, resampling=Resampling.nearest):
+    def reproject_match(
+        self, match_data_array, resampling=Resampling.nearest, **reproject_kwargs
+    ):
```
Comment (snowman2) [inline rioxarray/raster_array.py:461]: Mind adding `.. versionadded:: 0.9 reproject_kwargs` ?

File: rioxarray/raster_dataset.py [hunk 2]
```diff
@@ -129,7 +129,9 @@ def reproject(
                 resampled_dataset[var] = self._obj[var].copy()
         return resampled_dataset
 
-    def reproject_match(self, match_data_array, resampling=Resampling.nearest):
+    def reproject_match(
+        self, match_data_array, resampling=Resampling.nearest, **reproject_kwargs
+    ):
```
Comment (snowman2) [inline rioxarray/raster_dataset.py:134]: Mind adding `.. versionadded:: 0.9 reproject_kwargs` ?

File: rioxarray/raster_dataset.py [hunk 3]
```diff
@@ -147,7 +149,8 @@ def reproject_match(self, match_data_array, resampling=Resampling.nearest):
             Dataset with the target resolution and projection.
         resampling: rasterio.enums.Resampling, optional
             See :func:`rasterio.warp.reproject` for more details.
-
+        **reproject_kwargs:
+            Other options to pass to :obj:`rioxarray.raster_array.reproject()`
```
Comment (snowman2) [inline rioxarray/raster_dataset.py:153]:
```suggestion
            Other options to pass to :meth:`rioxarray.raster_array.RasterArray.reproject`
```

File: rioxarray/raster_array.py (hunk 1 above)
Comment (four43) [reply rioxarray/raster_array.py:461]: You got it - I'm not sure I did the comments quite right for the docs.

File: rioxarray/raster_array.py [hunk 4]
```diff
@@ -474,7 +476,8 @@ def reproject_match(self, match_data_array, resampling=Resampling.nearest):
             DataArray of the target resolution and projection.
         resampling: rasterio.enums.Resampling, optional
             See :func:`rasterio.warp.reproject` for more details.
-
+        **reproject_kwargs:
+            Other options to pass to :obj:`rioxarray.raster_array.reproject()`
```
Comment (snowman2) [inline rioxarray/raster_array.py:480]:
```suggestion
            Other options to pass to :meth:`rioxarray.raster_array.RasterArray.reproject`
```

File: rioxarray/raster_dataset.py (hunk 3 above)
Comment (four43) [reply rioxarray/raster_dataset.py:153]: Ah nice, just commented above on not knowing how the docs quite went. Thanks!

**Comment by snowman2** [comment]
This makes sense to me, thanks :+1: . Mind adding a simple test to ensure that the `nodata` passed into `reproject_match` works properly?

**Comment by four43** [comment]
@snowman2 how do you create these fixture files? When trying to re-use the same parameterized fixtures I keep getting this while trying to open mine:
```
ValueError: Given file dataset contains more than one data variable. Please read with xarray.open_dataset and then select the variable you want.
```

**Comment by four43** [comment]
I squashed and committed what I have but will need help with that fixture. Thanks!

**Comment by snowman2** [comment]
I am thinking you need to create the netCDF file starting from a DataArray instead of from a Dataset. Are you able to do that?

**Comment by four43** [comment]
I tried that as well, I gave it the ol:
```
mds_repr['__xarray_dataarray_variable__'].to_netcdf(modis_reproject_match__passed_nodata["compare"])
```
And `open_dataarray()` is still mad! I don't really know why.

**Comment by snowman2** [comment]
What happens if you add `decode_coords="all"` when opening?

**Comment by four43** [comment]
That did it! It's odd that it was required there for some reason...

**Comment by snowman2** [comment]
> It's odd that it was required there for some reason...

Related to #287

**Review by alfredoahds — APPROVED** [review]

**Comment by snowman2** [comment]
Thanks @four43 :+1:

**Comment by four43** [comment]
Thanks for the review and help @snowman2. We appreciate the project :D

**Pull request merged**